### DISCUSSION TITLE
GraphQL API rate limit for unauthenticated users?

### DISCUSSION THREAD

**Discussion by elenakrittik** [community/community#139866] [Apps, API and Webhooks] [answered]
### Select Topic Area

Question

### Body

[The GraphQL API docs](https://docs.github.com/en/graphql/overview/rate-limits-and-node-limits-for-the-graphql-api#primary-rate-limit) don't mention that, but [the REST API docs](https://docs.github.com/en/rest/using-the-rest-api/rate-limits-for-the-rest-api?apiVersion=2022-11-28#primary-rate-limit-for-unauthenticated-users) do (60 requests/hour).

**Comment by nangerno** [comment]
The GraphQL API and REST API have different rate limiting approaches for GitHub. 

- GitHub REST API: 60reqs/hr

- GitHub GraphQL API: 5,000 points/hr

**Reply by elenakrittik** [reply]
So, the GraphQL API rate limit for unauthenticated users is the same as for authenticated? That sound strange (and also doesn't match my experience, which was the reason i posted this question)

**Reply by nangerno** [reply]
As I mentioned in the previous section, the GitHub REST and GraphQL APIs have different rate limiting approaches. 
However, the GraphQL API cannot be used without authentication at all.
If authenticated, the GraphQL API has a rate limit of 5,000 points per hour. 
Is it clear for your case?

**Reply by elenakrittik** [reply]
It was clear to me that they use different scoring systems for rate limits. I only did the REST API analogy because it included a section that GraphQL API did not, not because i thought their scoring systems were the same. The ratelimit information your provided was already accessible in the exact pages i linked in my original post. Even though i figured it out earlier, thank you for clarifying that the GraphQL API can only be used with authentication, it is all clear now.

**Comment by elenakrittik** [comment, marked as answer]
Turns out the GraphQL API cannot at all be used without authenticating. A bummer.
